### PR TITLE
Link SQL exports to Django view

### PR DESCRIPTION
**Pull request by NoahCarnahan**
@benrudolph @snopoke cc @kaapstorm 
This PR adds a django view that can be used to download exports in the new format. This is a very minimal implementation, that is still missing a lot of functionality, like filters and case exports. The goal here was mainly to figure out which pieces are missing before everything can be wired together.

### CODE REVIEW

**Comment by snopoke** [comment]
Merged master to force travis build

File: corehq/apps/export/tasks.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from celery.task import task
+
+from corehq.apps.export.export import get_export_file
+from couchexport.models import Format
+from couchexport.tasks import escape_quotes
+from soil.util import expose_cached_download
+
+
+@task
+def get_export_download(export_instances, filters, download_id, filename=None, expiry=10*60*60):
```
Comment (dimagimon) [inline corehq/apps/export/tasks.py]: E226 missing whitespace around arithmetic operator

File: corehq/apps/export/views.py [hunk 2]
```diff
@@ -1417,3 +1429,34 @@ def commit(self, request):
...
+
+    def _get_download_task(self, export_specs, export_filter, max_column_size):
+        # TODO: Actually do something with the arguments...
+        # TODO: Don't assume form!
+        export_instance = FormExportInstance.get(self.export_id)
+        self._check_deid_permissions(export_instance)
+
+        return get_download_task(
+            export_instances=[export_instance],
+            filters=[], # TODO: Do something with export_filters
```
Comment (dimagimon) [inline corehq/apps/export/views.py]: E261 at least two spaces before inline comment

**Comment by NoahCarnahan** [comment]
@snopoke updated

**Comment by benrudolph** [comment]
:+1: :mans_shoe: :construction:

File: corehq/apps/export/tests/test_export_forms.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    def test_get_group_filter(self):
+        raise NotImplementedError
+
+    @patch('corehq.apps.reports.util.get_first_form_submission_received', lambda x: datetime.datetime(2015, 1, 1))
+    def test_get_user_filter(self):
+        form_data = {'type_or_group': 'type', 'group': '', 'user_types': ['mobile', 'unknown']}
+        form = FilterFormESExportDownloadForm(Domain(name="foo"), pytz.utc, form_data)
+        self.assertTrue(form.is_valid())
+        raise NotImplementedError
+
```
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_forms.py]: W391 blank line at end of file

**Comment by NoahCarnahan** [comment]
Made some adjustments to the views to use the new filter classes. Didn't have time to finish the tests this afternoon, but will work on them monday.

File: corehq/apps/export/tests/test_export_forms.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        DOES NOT verify that BaseFilterExportDownloadForm._get_filtered_users()
+        retrieves the correct user ids from the database.
+        """
+        form_data = {
+            'type_or_group': 'type',
+            'user_types': ['mobile', 'unknown'],
+            'group': '',
+            'date_range': '2015-06-25 to 2016-02-19',
+        }
+        def mock_users_matching_filter(domain, user_filters):
```
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_forms.py]: E301 expected 1 blank line, found 0

File: corehq/apps/export/tests/test_export_forms.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            return [None, "some_user_id", "some_other_user_id"]
+
+        form = FilterFormESExportDownloadForm(Domain(name="foo"), pytz.utc, form_data)
+
+        self.assertTrue(form.is_valid(), "Form had the following errors: {}".format(form.errors))
+        with patch("corehq.apps.export.forms.users_matching_filter", mock_users_matching_filter):
+            user_filter = form._get_user_filter()
+        self.assertEqual(user_filter.submitted_by, [None, "some_user_id", "some_other_user_id"])
+
```
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_forms.py]: W391 blank line at end of file

**Comment by NoahCarnahan** [comment]
:white_check_mark:

File: corehq/apps/export/tests/test_export_forms.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
+import datetime
+from collections import namedtuple
+
+import pytz
+from django.test import SimpleTestCase
+from mock import patch
+
+from corehq.apps.domain.models import Domain
```
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_forms.py]: F401 'Domain' imported but unused

File: corehq/apps/export/tests/test_export_forms.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+from corehq.apps.export.filters import FormSubmittedByFilter
+from corehq.apps.export.forms import FilterFormESExportDownloadForm
+
+
+@patch('corehq.apps.reports.util.get_first_form_submission_received', lambda x: datetime.datetime(2015, 1, 1))
+class TestFilterFormESExportDownloadForm(SimpleTestCase):
+
+    def setUp(self):
+        DomainObject = namedtuple('DomainObject', ['uses_locations', 'name'])
+        self.project =  DomainObject(False, "foo")
```
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_forms.py]: E222 multiple spaces after operator

**Comment by benrudolph** [comment]
sweet, maybe just cleanup the last dimagimon and then good from my end

**Comment by NoahCarnahan** [comment]
(fixed)

**Pull request merged**